modules/apps/object/object-service/src/main/java/com/liferay/object/internal/sort/ObjectEntryFieldSortDSLQueryVisitor.java

```diff
@@ -77,7 +78,9 @@ public DSLQuery visit(DSLQuery dslQuery, Sort sort) throws PortalException {
 		}
 
 		OrderByExpression orderByExpression = _getOrderByExpression(
-			_isParentComplexField(sort), columnExpression, sort.isReverse());
+			_isParentComplexField(sort), columnExpression, sort.isReverse(),
+			Objects.equals(objectField.getDBType(), "Boolean") ?
```

Comment:
There is a constant for this purpose: `ObjectFieldConstants.DB_TYPE_BOOLEAN`

Reply:
Fixed and push forced.